base/experimental.jl

```diff
@@ -658,9 +658,9 @@ function wait_with_timeout(c::GenericCondition; first::Bool=false, timeout::Real
             # Confirm that the waiting task is still in the wait queue and remove it. If
             # the task is not in the wait queue, it must have been notified already so we
             # don't do anything here.
-            if !waiter_left[] && ct.queue == c.waitq
+            if !waiter_left[] && ct.queue === c.waitq
                 dosched = true
-                Base.list_deletefirst!(c.waitq, ct)
+                Base.list_deletefirst!(c.waitq::IntrusiveLinkedList{Task}, ct)
```

Comment:
Why is this necessary?
```julia
julia> fieldtypes(Base.ThreadSynchronizer)
(Base.IntrusiveLinkedList{Task}, Base.Threads.SpinLock)
```

Reply:
Just to follow the pattern elsewhere mostly.

Reply:
What pattern?